src/lib/registration/ui/not_installed_products_dialog.rb

```diff
@@ -0,0 +1,235 @@
+          log.info "User input: #{ui}"
+          case ui
+          when :install
+            not_installed = install_products
+            # TRANSLATORS: Popup error showing all the addons that weren't
+            # installed, %s is the addons identifiers.
+            Yast::Popup.Error(_("These addons were not installed:\n\n%s") %
+                              not_installed.join("\n")) unless not_installed.empty?
+            update_summary
+          when :sync
+            registration_ui.synchronize_products(SwMgmt.installed_products)
+            update_summary
+          when :next, :cancel
+            return ui
+          end
+
+          return :next if !registered_not_installed_addons?
+        end
+      end
+
+      # Install the not installed products
+      # @return [Array] list of not installed (failed) products, empty list on success
+      def install_products
+        not_installed = []
+
+        Addon.registered_not_installed.each do |addon|
+          # FIXME: fix the product installation, the add-on name might not match
+          # the libzypp product name
+          # Check for possible conflicts and let the user solve them,
+          # confirm package licenses if there are any,
+          # and run the package installation
+          if addon_product_installable?(addon)
+
+            result = Yast::Pkg.PkgCommit(0)
+            # success?
+            if result && result[1].empty?
+              # FIXME: This method is not available in SLE-12-GA
+              # Yast::PackagesUI.show_update_messages(result)
+              next
+            end
+            #
+            next if result && result[1].empty?
+          end
+
+          log.error("Product #{addon.identifier} could not be installed")
+          # revert the changes
+          Yast::Pkg.PkgApplReset
+          Yast::Pkg.PkgReset
+          not_installed << addon.identifier
+        end
+
+        not_installed
+      end
+
+      # RichText summary of the installed but not registered products.
+      #
+      # @return [String]
+      def not_installed_addons_summary
+        # TRANSLATORS: A RichText warning about all the products registered but
+        #   not installed. (1/2)
+        summary = _("<p>The addons listed below are registered but not installed: </p>")
+
+        summary << "<ul>#{not_installed_addon_names.map { |a| "<li>#{a}</li>" }.join("")}</ul>"
+
+        # TRANSLATORS: A RichText warning about all the products registered but
+        #   not installed. (2/2)
+        summary << _("<p><b>Synchronize</b> your products if you want to <b>deactivate</b> " \
+                     "them at your registration server.</p>")
+
+        summary
+      end
+
+      # It updates the summary of registered but not installed products
+      # updating also Addon cache.
+      def update_summary
+        log.info "Updating summary"
+        Yast::Popup.Feedback(RegistrationUI::CONTACTING_MESSAGE, _("Updating summary")) do
```

Comment:
You can reuse the proposed "Checking registration status" label here. Save the translators work, and also we do not have to reveal all internals to the user.